tests/test_inference.py

```diff
@@ -774,6 +774,36 @@ def test_three_sites(self):
         self.verify([[1, 1], [1, 1], [1, 1]])
 
 
+class TestTimeUnits:
+    """
+    Tests that time units get set properly
+    """
+
+    def test_normally_uncalibrated(self):
+        ts = msprime.simulate(10, mutation_rate=5, recombination_rate=4, random_seed=21)
+        assert ts.time_units != tskit.TIME_UNITS_UNCALIBRATED
```

Comment:
Might as well make this concrete, ``ts.time_units == "generations"`` as it's not going to change.